tools/stats/filtering.py

```diff
@@ -186,7 +186,7 @@ def stop_err(msg):
 # Work out which columns are used in the filter (save using 1 based counting)
 used_cols = sorted(set(int(match.group()[1:])
                    for match in re.finditer(r'c(\d)+', cond_text)))
-largest_col_index = max(used_cols)
+largest_col_index = min(in_columns, max(used_cols))
```

Comment:
This will lead to everything being invalid lines (but a green dataset) since the filter condition can never be fulfilled. I'm not sure that's always the correct thing to do. It seems like we should error out by default when this happens (in most cases I assume this is not intended), with a tool option to output an empty file if that's a legitimate outcome. Does that seem reasonable ?

Reply:
The configuration options should be something like:
Yes/No: Allow empty datasets
Select:
- All expected columns must be present
- Filter records with fewer than expected columns
- Output records with fewer than expected columns

Reply:
Yep, that sounds good to me.